### ISSUE TITLE
Confirmation Interaction issue

### ISSUE THREAD

**Issue by ustaalon** [LethalCompany/LethalAPI.TerminalCommands#18]
Hey guys,
So I wrote my first mod based on this repo, and I'm not sure if I found a bug.

When using [ConfirmationInteraction](https://github.com/LethalCompany/LethalAPI.TerminalCommands/blob/development/LethalAPI.TerminalCommands/Interactions/ConfirmInteraction.cs), eventually it will use this logic:
```cs
		/// <summary>
		/// Handles terminal input for the terminal interaction
		/// </summary>
		/// <param name="arguments">Arguments provided by the user</param>
		/// <returns>Interaction response, or <see langword="null"/></returns>
		public object HandleTerminalResponse(ArgumentStream arguments)
		{
			if (!arguments.TryReadNext(out string response))
			{
				return Deny(arguments);
			}

			if (response.Trim().Equals("confirm", StringComparison.InvariantCultureIgnoreCase))
			{
				return Confirm(arguments);
			}

			return Deny(arguments);
		}
```
The last Deny() in the function will be invoked automatically at the confirmation interface (before the user decides to deny or not). I solved it by doing this:

```cs

        /// <summary>
        /// Handles terminal input for the terminal interaction
        /// </summary>
        /// <param name="arguments">Arguments provided by the user</param>
        /// <returns>Interaction response, or <see langword="null"/></returns>
        public object HandleTerminalResponse(ArgumentStream arguments)
        {
            if (!arguments.TryReadNext(out string response))
            {
                return Deny(arguments);
            }

            if (response.Trim().Equals("confirm", StringComparison.InvariantCultureIgnoreCase))
            {
                return Confirm(arguments);
            }
            else if (response.Trim().Equals("deny", StringComparison.InvariantCultureIgnoreCase))
            {
                return Deny(arguments);
            }

            return null;
        }
```

I think I found some more issues but I can't say for sure as this is my first time handling with mods. 

With that said, you did an awesome job!

**Comment by ShimmyMySherbet** [comment]
> The last Deny() in the function will be invoked automatically at the confirmation interface (before the user decides to deny or not). I solved it by doing this:

Something to keep in mind here, is once that code block runs, the user has already responded. This is not the code the prompts the user for a response, but the code that handles the response the user entered.

If the user did not enter anything, it denies it at the first block. If he user explicitly typed 'confirm', it confirms it at the second block. Anything else gets denied.

Also, returning null here would cascade the terminal interaction response to the command handler, to be executed as a command.

If I have misunderstood the issue here, please correct me, but at the moment I'm not seeing the problem.

> So I wrote my first mod based on this repo

Did you copy this repo's code into your mod, or are you relying on this repo's DLL?

Copying the code into your own mod would work for that mod alone, but would break comparability with other mods using the API. Usually you would just reference this mod's dll, and then set this mod as a dependency on Thunderstore. However this library hasn't yet been released on thunderstore.

**Comment by ustaalon** [comment]
Hey @ShimmyMySherbet thanks for the reply! Because I did ConfirmationInteraction with Confirm and Deny action, entering this interface automatically invoked the deny action. Therefore the user saw the deny response and not the interface that he should reply to.

I also had issues with the invoke of the confirm/deny, but I was able to do some tweaks. But because I took the code and added it to my repo, it probably will have some issues as you said. (I change the namespaces so it should not affect I believe), But I can use the DLL itself of the current version and/or wait for it to be published in Thunderstorm.

Either way, I added you in Discord, maybe we can talk over the weekend and I will show you the issues I faced.
